File: socket/discu/message.go
package discu

import "time"

type Message struct {
	Author		string `json:"author"`
	Body			string `json:"body"`
	Channel		string `json:"channel"`
	TimeSent	time.Time
}

func (self *Message) String() string {
	return self.Author + " says " + self.Body + " in " + self.Channel
}

File: web/root/js/models/message.js
var app = app || {};

// Message Model
app.Message = function(data) {
  this.author = m.prop(data.author);
  this.body = m.prop(data.body);
  this.timesent = m.prop(data.timesent);
  this.channel = m.prop(data.channel);
};

var list = [];
app.List = function() {
  return list;
};

app.Channel = function(data) {
  this.name = data.name
  this.messages = [];
};
